File: Problems/0002_Add_Two_Number/Project_Scala/Operate_ListNode.scala
object Operate_ListNode {
    def set_ListNode(nums:Array[Int]):ListNode = {
        return set_ListNode(nums, 0)
    }

    def set_ListNode(nums:Array[Int], index:Int):ListNode = {
        if (index >= nums.size)
            return null
        
        var node:ListNode = new ListNode(nums(index))
        node.next = set_ListNode(nums, index + 1)

        return node
    }

    def output_ListNode(ll:ListNode):String = {
        var retStr:String = ll.x.toString

        if (ll.next != null)
            retStr += " -> " + output_ListNode(ll.next)

        return retStr
    }
}

File: Problems/0024_Swap_Nodes_in_Pairs/Project_CS/Operate_ListNode.cs
using System;

public class Operate_ListNode
{
    public ListNode set_ListNode(int[] nums)
    {
        return set_ListNode(nums, 0);
    }

    public ListNode set_ListNode(int[] nums, int index)
    {
        if (nums == null)
            return null;

        if (index >= nums.Length)
            return null;
        
        ListNode node = new ListNode(nums[index]);
        node.next = set_ListNode(nums, index + 1);

        return node;
    }

    public string output_ListNode(ListNode node)
    {
        if (node == null)
            return "";

        string resultStr = node.val.ToString();

        if (node.next != null)
            resultStr += " -> " + output_ListNode(node.next);

        return (resultStr);
    }
}
